jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8307,13 +8307,71 @@ public String resolveConstantDesc(MethodHandles.Lookup lookup) {
 	}
 	
 	/**
-	 * Apply a function to this string. The function expects a single String input and returns an R.
+	 * Apply a function to this string. The function expects a single String input
+	 * and returns an R.
 	 * 
-	 * @param f - the functional interface to be applied
+	 * @param f 
+	 *			  the functional interface to be applied
+	 *
 	 * @return the result of application of the function to this string
+	 * 
+	 * @since 12
 	 */
 	public <R> R transform(Function<? super String, ? extends R> f) {
 		return f.apply(this);
 	}
+
+	/**
+	 * Indents each line of the string depending on the value of n, and normalizes
+	 * line terminators to the newline character "\n".
+	 * 
+	 * @param n 
+	 *			  the number of spaces to indent the string
+	 *
+	 * @return the indented string with normalized line terminators
+	 * 
+	 * @since 12
+	 */
+	public String indent(int n) {
+		Stream<String> lines = lines();
+		Iterator<String> iter = lines.iterator();
+		StringBuilder builder = new StringBuilder();
+
+		while (iter.hasNext()) {
+			String currentLine = iter.next();
+
+			if (n > 0) {
+				int spaceOffset = n + currentLine.length();
+				currentLine = String.format("%" + spaceOffset + "s", currentLine);
```

Comment:
Does this match the RI's behaviour?  Some jshell examples show that the RI is prepending a fixed amount of space:
```
jshell> "ab".indent(1)
$28 ==> " ab\n"

jshell> " ab".indent(1)
$29 ==> "  ab\n"
```
Note, the second example starts with a space and indent(1) still adds a space.

If the method should always add some number of spaces, the code create the string of spaces using String.repeat() and prepend that string always.